daemon/network.go

```diff
@@ -468,3 +468,35 @@ func (daemon *Daemon) deleteNetwork(networkID string, dynamic bool) error {
 func (daemon *Daemon) GetNetworks() []libnetwork.Network {
 	return daemon.getAllNetworks()
 }
+
+// clearAttachableNetworks removes the attachable networks
+// after disconnecting any connected container
+func (daemon *Daemon) clearAttachableNetworks() {
+	var networks []libnetwork.Network
+	for _, n := range daemon.GetNetworks() {
+		if n.Info().Attachable() {
+			networks = append(networks, n)
+		}
+	}
+	for _, n := range networks {
+		nwName := n.Name()
+		for _, ep := range n.Endpoints() {
+			epInfo := ep.Info()
+			if epInfo == nil {
+				continue
+			}
+			sb := epInfo.Sandbox()
+			if sb == nil {
+				continue
+			}
+			containerID := sb.ContainerID()
+			if err := daemon.DisconnectContainerFromNetwork(containerID, nwName, true); err != nil {
```

Comment:
Not necessarily. This API can take either name of ID (it eventually calls `FindNetwork()`), it _was_ in fact the first function which processes a network disconnect POST request.  I am saying "it was" because later with the addition of swarm networks being lazily programmed  `FindNetwork` had to be called before anyway.

In this case the network is known to this node and there is no need to pass the ID.

But I agree it will spare the reader from wondering about this logic when reading the code, I will change it to ID.

Thanks.